controller/class.tapestry-controller.php

```diff
@@ -259,9 +267,16 @@ public function updateTapestrySettings($settings)
         // TODO: add validation for the $settings
 
         $tapestry = get_post_meta($this->postId, 'tapestry', true);
+
+        if (!isset($tapestry)) {
+            $tapestry =  (object)array();
+        }
+
         $tapestry->settings = $settings;
 
-        $this->_updatePost($tapestry, 'tapestry');
+        if ($updateTapestryPost) {
```

Comment:
It's in `tapestry.php`. We would like to use this function when a tapestry is created but we don't want to add/update the same post again because it's already done by the WP interface/system.